pandas/tests/groupby/test_reductions.py

```diff
@@ -389,6 +391,37 @@ def test_groupby_non_arithmetic_agg_int_like_precision(method, data):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("how", ["first", "last"])
+def test_first_last_skipna(any_real_nullable_dtype, sort, skipna, how):
+    # GH#57019
+    if is_extension_array_dtype(any_real_nullable_dtype):
+        na_value = Series(dtype=any_real_nullable_dtype).dtype.na_value
+    else:
+        na_value = np.nan
```

Comment:
Not sure if there is any better way to get the NA value for a dtype when the code needs to span NumPy and EAs (both masked and pyarrow). This is the reason why I went with the string aliases in the `any_real_nullable_dtype` fixture for pyarrow; this is at odds with some of the other fixtures but the code to the NA value was much worse with pyarrow dtype objects.

cc @mroeschke if you have any suggestions

Reply:
Use can use `pandas_dtype` to get a dtype object from the string and `na_value_for_dtype` to get the na value from the dtype object

Reply:
Also I think we've been using `isinstance(..., ExtensionDtype)` instead of `is_extension_array_dtype` if possible

Reply:
> I just don't want to introduce an inconsistency here (pyarrow dtype objects vs string alias) if it's avoidable.

Yeah I don't think it's avoidable as of now, so I'm okay the way you have it in this PR